src/uct/ib/base/ib_iface.h

```diff
@@ -554,6 +555,14 @@ uint8_t uct_ib_iface_config_select_sl(const uct_ib_iface_config_t *ib_config);
     uct_ib_device_name(uct_ib_iface_device(_iface)), (_iface)->config.port_num
 
 
+#define UCT_IB_IFACE_CHECK_CAP_LIMIT(_iface, _caps, _name, _limit) \
+    if ((_caps)->_name > IBV_DEV_ATTR_FIELD(&(_iface)->dev_attr, _limit)) { \
+        ucs_debug("iface=%p: reduced  from %d to %d based on caps", \
```

Comment:
1. can it be regular function? why does it have to be a macro?
2. pls also print the string name of what is being reduced 
`"iface %p: reduced %s from %d to %d based on caps"`